graylog2-web-interface/src/components/graylog/Button.jsx

```diff
@@ -1,19 +1,19 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import { Button as BootstrapButton } from 'react-bootstrap';
 import styled from 'styled-components';
 
 import buttonStyles from './styles/button';
 import { propTypes, defaultProps } from './props/button';
 
-const Button = ({ active, bsStyle, ...props }) => {
-  const StyledButton = styled(BootstrapButton)`
+const Button = React.forwardRef(({ active, bsStyle, ...props }, ref) => {
+  const StyledButton = useCallback(styled(BootstrapButton)`
```

Comment:
Would it help avoiding `useCallback` if we would move the `StyledButton` declaration to the top scope of the file and make it look like this:

```
const StyledButton = ({ active }) => styled(BootstrapButton)`${buttonStyles({ active })`;
```

Same for the following usages of `useCallback`.

Reply:
@dennisoelkers We are unable to move the `StyledButton` creation outside of the component declaration because the theme stuff is all using hooks which "Hooks can only be called inside of the body of a function component."